### PR TITLE
Interact Button in Documentation

### PR DESCRIPTION
**Pull request by smithis7**
**Description**

Adds a button in the documentation that allows you to launch binder.

**Examples**

See this page for an example: https://smithis7.github.io/tardis/branch/interact_badge_doc/quickstart/quickstart.html

**Type of change**

- [ ] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [x] I have updated the documentation accordingly.
    - [x] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: docs/conf.py
Comment (jaladh-singhal) [inline docs/conf.py:127]:
```suggestion
    <style>
        .launch-btn {
            background-color: #2980B9;
            border: none;
            border-radius: 4px;
            color: #fcfcfc;
            font-family: inherit;
            text-decoration: none;
            padding: 3px 8px;
            letter-spacing: 0.03em;
            display: inline-block;
            line-height: 1.5em;
        }

        .launch-btn:hover {
            background-color: #1b6391;
        }

        .launch-btn:visited {
            color: #fcfcfc;
        }

        .note-p {
            margin-bottom: 0.4em;
            line-height: 2em;
        }
    </style>
    
    <div class="admonition note">
    <p class="note-p">You can interact with this notebook online: <a href="https://mybinder.org/v2/gh/tardis-sn/tardis/HEAD?filepath={{ docname|e }}" class="launch-btn" target="_blank" rel="noopener noreferrer">Launch Interactive Version</a></p>
    <p>And you can download it from
<a class="reference external" href="https://github.com/tardis-sn/tardis/tree/master/docs/quickstart/quickstart.ipynb" target="_blank" rel="noopener noreferrer">GitHub</a></p>
    </div>
```

File: docs/conf.py
Comment (jaladh-singhal) [inline docs/conf.py:140]:
```suggestion
            background-color: #1b6391;
            color: #fcfcfc;
```
The text color turns blue on hover that shouldn't be the case (only background color should darken) - this will fix it.

**Review by jaladh-singhal — CHANGES_REQUESTED** [review]
Looks great - there's one thing that I missed last time.

**Review by jaladh-singhal — APPROVED** [review]
Looks as expected! Thanks for changes.

**Review by andrewfullard — APPROVED** [review]

**Comment by epassaro** [comment]
My opinion is we should take care of the aesthetic of the documentation. Personally, I don't like the changes so much (specially the second line of text). Also, adding a block of CSS code seems overcomplicated for a task that should be easy to solve with shields.io .

Anyway, the button works as expected and that's not an easy job! Congrats. Feel free to merge this if you are satisfied with the results.

**Comment by DhruvSondhi** [comment]
This addition seems to be good as we are trying to integrate Binder with TARDIS code notebooks. I would agree that with @epassaro on the fact that this CSS needs to be added in the `nbsphinx prolog`. `Prologs` & `Epilogs` allow us to add anything in the `r"""` & it will be appended before the main documentation (only where `nbsphinx` is working). Keeping that in mind, the page may become unresponsive or laggy if the load is increased via Binder via the button.
Another thing is that there should be a way to access the Binder Repository from the Main Documentation page such that the user should not be just navigating the documentation to find a Page that has this button & then click on it to access it via Binder. It should allow the user to check for all the possible Binder Pages that can be accessed.

**Comment by jaladh-singhal** [comment]
> My opinion is we should take care of the aesthetic of the documentation.

@epassaro We used the same white shade text on blue shade background as on top left (where tardis home is mentioned). Can you please elaborate what do you think that can be improved?

> Personally, I don't like the changes so much (specially the second line of text). 

@epassaro Where do you think is better to put second line of text?

> Also, adding a block of CSS code seems overcomplicated for a task that should be easy to solve with shields.io

That's true but for some users it's not identifiably clickable as @smithis7 pointed out - so I have to write CSS to create an actual button.

File: docs/conf.py
Comment (epassaro) [inline docs/conf.py:157]: points to quickstart!

Comment (smithis7) [reply docs/conf.py:157]: Didn't catch that.... whoops

**Comment by jaladh-singhal** [comment]
>  I would agree that with @epassaro on the fact that this CSS needs to be added in the nbsphinx prolog
@DhruvSondhi it's already added in prolog - check the code. Did you mean something else - let me know if there's better place to put it?

**Comment by epassaro** [comment]
> > My opinion is we should take care of the aesthetic of the documentation.
> 
> @epassaro We used the same white shade text on blue shade background as on top left (where tardis home is mentioned). Can you please elaborate what do you think that can be improved?
> 
> > Personally, I don't like the changes so much (specially the second line of text).
> 
> @epassaro Where do you think is better to put second line of text?
> 
> > Also, adding a block of CSS code seems overcomplicated for a task that should be easy to solve with shields.io
> 
> That's true but for some users it's not identifiably clickable as @smithis7 pointed out - so I have to write CSS to create an actual button.

I found the choice of words redundant (interact-interactive, you can x2), not sure about using a semicolon before the button. Also, I'm not a fan of title case, but I'm not a native English speaker so probably it's just in my head.

```
You can interact with this notebook online: Launch Interactive Version

And you can download it from GitHub.
```

About the second line: I found it too short, not worthing a new line.

**Comment by andrewfullard** [comment]
Are we actually happy with this? @epassaro do you want us to wait for further discussion before merging?

**Comment by jaladh-singhal** [comment]
> I found the choice of words redundant (interact-interactive, you can x2)

Accounted, also made clear the benefit of launching interactive version is no need of installing locally if user is interested after playing with it then they can.

> not sure about using a semicolon before the button. 

It's because it's an inline button - I tried without colon, that looks more odd

> Also, I'm not a fan of title case, but I'm not a native English speaker so probably it's just in my head.

I can understand it may appear to loud to some users, changed to sentence style - I missed it rule 5 in [the article I shared](https://uxmovement.com/buttons/5-rules-for-choosing-the-right-words-on-button-labels/) also says so.

> About the second line: I found it too short, not worthing a new line.

Accounted made it bit descriptive and complimentary to 1st line. And in same line it looks too cluttered - also bear in mind that we will reduce width of main page container to 800px (#1687) in coming days (so there won't be that much white space).

@epassaro @smithis7 check this out with all changes:
image

**Review by epassaro — APPROVED** [review]

**Review by jaladh-singhal — CHANGES_REQUESTED** [review]
@smithis7 can you make the changes in code as per screenshot I shared (they're mostly textual edits)?

**Review by jaladh-singhal — APPROVED** [review]

**Pull request merged**
